server/services/emailNotifier.js

```diff
@@ -0,0 +1,154 @@
+import nodemailer from "nodemailer";
+import Mailgen from "mailgen";
+import dotenv from "dotenv";
+import User from "../models/User.js";
+import Chore from "../models/Chore.js";
+
+dotenv.config();
+
+export const startEmailService = async () => {
+  let today = new Date(); // get today's date
+
+  // set the UTS to 12am
+  const millisecondsPerDay = 86400000;
+  const unixEpochMilliseconds = Date.parse("1970-01-01T00:00:00Z");
+  today =
+    Math.floor((today * 1000) / millisecondsPerDay) * millisecondsPerDay +
+    unixEpochMilliseconds;
+
+  const toDaysSelector = { days: 1, weeks: 7, month: 30, year: 365 };
+
+  const dateSelector = {
+    today: today,
+    nextOccurrence: undefined,
+  };
+
+  let config = {
+    service: "gmail",
+    auth: {
+      user: process.env.EMAIL,
+      pass: process.env.EMAIL_PASSWORD,
+    },
+  };
+
+  let transporter = nodemailer.createTransport(config);
+  let MailGenerator = new Mailgen({
+    theme: "cerberus",
+    product: {
+      name: "Chorekeeper",
+      link: process.env.FRONTEND, //add frontend link
+    },
+  });
+
+  try {
+    //get user
+    let users = await User.find({});
+
+    // for loop to loop through all the users
+    for (let i = 0; i < users.length; i++) {
+      let user = users[i];
+      //collect user chores
+      let chores = await Chore.find({
+        _id: { $in: user.chores },
+        nextOccurrence: { $exists: true },
+      });
+
+      //select todays chores
+      let tempChores = chores.filter((chore) => {
+        let timeframeIndays =
+          toDaysSelector[chore.frequency.interval] * chore.frequency.quantity;
+
+        //approximate next occurrence to nextOccurrence at 12am
+        dateSelector.nextOccurrence =
+          Math.floor((chore.nextOccurrence * 1000) / millisecondsPerDay) *
+            millisecondsPerDay +
+          unixEpochMilliseconds;
+
+        return (
+          (dateSelector.today - dateSelector.nextOccurrence) %
+            timeframeIndays ===
+          0
+        );
+      });
+      let todaysChores = tempChores.map((chore) => {
+        return {
+          Chore: chore.name,
+          Location: chore.location,
+          "Duration(minutes)": chore.duration / 60,
+        };
+      });
+
+      //select overdue chores
+      let overChores = chores.filter((chore) => {
+        //approximate next occurrence to nextOccurrence at 12am
+        dateSelector.nextOccurrence =
+          Math.floor((chore.nextOccurrence * 1000) / millisecondsPerDay) *
+            millisecondsPerDay +
+          unixEpochMilliseconds;
+        return dateSelector.nextOccurrence < dateSelector.today;
+      });
+      let overdueChores = overChores.map((chore) => {
+        return {
+          Chore: chore.name,
+          Location: chore.location,
+          "Duration(minutes)": chore.duration / 60,
+          // Preference: chore.preference,
+          From: new Date(chore.nextOccurrence).toLocaleDateString("en-US"),
```

Comment:
I think if we don't specify en-US, it will just convert to whatever locale the user is in, which may be nice for them.